### PR TITLE
Switch from std::tuple to evolution::dg::BoundaryData class

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- This just makes changing the members of this class way easier.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DiscontinuousGalerkin/BoundaryData.hpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * 1. the mesh of the ghost cell data we received. This allows eliding
+ *    projection when all neighboring elements are doing DG.
+ * 2. the mesh of the neighboring element's face (not the mortar mesh!)
+ * 3. the variables at the ghost zone cells for finite difference/volume
+ *    reconstruction
+ * 4. the data on the mortar needed for computing the boundary corrections (e.g.
+ *    fluxes, characteristic speeds, conserved variables)
+ * 5. the TimeStepId beyond which the boundary terms are no longer valid, when
+ *    using local time stepping.
+ * 6. the troublade cell indicator status using for determining halos around
```
Comment (knelli2) [inline src/Evolution/DiscontinuousGalerkin/BoundaryData.hpp]: `troublade` ⚔️

Comment (nilsdeppe) [reply src/Evolution/DiscontinuousGalerkin/BoundaryData.hpp]: 🤣 the truest of all blades :D 🗡️

Comment (nilsdeppe) [reply src/Evolution/DiscontinuousGalerkin/BoundaryData.hpp]: Okay, squashed in a fix 🤞

File: src/Evolution/DiscontinuousGalerkin/BoundaryData.hpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * 1. the mesh of the ghost cell data we received. This allows eliding
+ *    projection when all neighboring elements are doing DG.
+ * 2. the mesh of the neighboring element's face (not the mortar mesh!)
+ * 3. the variables at the ghost zone cells for finite difference/volume
+ *    reconstruction
+ * 4. the data on the mortar needed for computing the boundary corrections (e.g.
+ *    fluxes, characteristic speeds, conserved variables)
+ * 5. the TimeStepId beyond which the boundary terms are no longer valid, when
+ *    using local time stepping.
+ * 6. the troubled cell indicator status using for determining halos around
```
Comment (kidder) [inline src/Evolution/DiscontinuousGalerkin/BoundaryData.hpp]: using -> used

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

File: src/Evolution/DiscontinuousGalerkin/BoundaryData.hpp (hunk 2 above)
Comment (nilsdeppe) [reply src/Evolution/DiscontinuousGalerkin/BoundaryData.hpp]: Done!

**Review by kidder — APPROVED** [review]

**Pull request merged**